torch/csrc/dynamo/eval_frame.c

```diff
@@ -304,10 +305,49 @@ typedef struct cache_entry {
   struct cache_entry* next;
 } CacheEntry;
 
+#define Py_NONE_CACHE_ENTRY (CacheEntry*)Py_None
+
+static void destroy_cache_entry(CacheEntry* e);
+
+#define DECLARE_CACHE_ENTRY_ATTR(name) \
+static PyObject* CacheEntry_##name(CacheEntry* self, PyObject* _noargs) { \
+  PyObject* res = (PyObject*)self->name; \
+  Py_XINCREF(res); \
+  return res; \
+}
+
+DECLARE_CACHE_ENTRY_ATTR(check_fn)
+DECLARE_CACHE_ENTRY_ATTR(code)
+DECLARE_CACHE_ENTRY_ATTR(next)
+
+static struct PyGetSetDef CacheEntry_properties[] = {
+    {"check_fn", (getter)CacheEntry_check_fn, NULL, NULL, NULL},
+    {"code", (getter)CacheEntry_code, NULL, NULL, NULL},
+    {"next", (getter)CacheEntry_next, NULL, NULL, NULL},
+    {NULL}};
+
+static PyTypeObject CacheEntryType = {
+  PyVarObject_HEAD_INIT(NULL, 0)
+  .tp_name = "torch._C.dynamo.eval_frame.CacheEntryWrapper",
+  .tp_basicsize = sizeof(CacheEntry),
+  .tp_itemsize = 0,
+  .tp_flags = Py_TPFLAGS_DEFAULT,
+  .tp_new = PyType_GenericNew,
```

Comment:
Using GenericNew is probably not the right call, since you have fields on cache_entry which need to be initialized first, so if you do it as you've done here, if you call a getter after you have constructed the object you will segfault because the fields are NULL.

Reply:
Good point. Added a `tp_new` and `tp_init`.